Select APN based on ICCID (#85)
nerves-networking/vintage_net_qmi: README.md

Currently only one service provider is supported, so replace `"fill_in"` with
the APN that they gave you.

> ## APN selection via ICCID prefixes
> 
> Optionally, you can configure the service provider configuration to have a list
> of ICCID prefixes that will be checked against the ICCID of the SIM card in
> order to select the correct APN.
> 
> If a service provider is not configured to be selected against an ICCID it will
> be the default APN used when trying to connect.

Reviewer: ```suggestion
## Service provider APN selection via ICCID

If you're deploying SIMs from multiple service providers, you'll need to set
the APN based on what SIM was installed in the device. One way is to set
a configuration manually when you install the SIM. An easier way is to have
`VintageNetQMI` pick the APN based on the SIM's ICCID.

This featured is enabled by using the `:only_iccid_prefixes` option and listing
multiple service providers. `VintageNetQMI` looks at the `:service_providers` 
list in order until one matches. Leave off the`:only_iccid_prefixes` option on
the final service provider to provide a catch-all APN.
```

> The example below will check the prefixes against the ICCID read from the SIM
> card to see if the APN provider should be used. If the SIM card's ICCID prefix
> does not match either option `VintageNetQMI` will not try to connect as there is
> default provided:
> 
> ```elixir
> VintageNet.configure("wwan0", %{
>       type: VintageNetQMI,
>       vintage_net_qmi: %{
>         service_providers: [%{apn: "the_apn", only_iccid_prefixes: ["8973611", "8973612"]}]
>       }
>     })
> ```
> 
> To fix this we can provide a service provider configuration without any prefix
> configuration. This is will become the default APN that `VintageNetQMI` trys to
> connect to if the SIM card's ICCID does not match the prefixes provided for other
> APNs.

Reviewer: ```suggestion
Here's an example:
```

```elixir
VintageNet.configure("wwan0", %{
      type: VintageNetQMI,
      vintage_net_qmi: %{
        service_providers: [

>           %{apn: "the_apn", only_iccid_prefixes: ["8973611", "8973612"]},
>           %{apn: "default apn"}

Reviewer: ```suggestion
          %{apn: "special_service_provider", only_iccid_prefixes: ["8973611", "8973612"]},
          %{apn: "default_apn"}
```
